trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/IMemoryAdressing.java

```diff
@@ -0,0 +1,251 @@
+	 *
+	 * @return The statements.
+	 */
+	List<Statement> getChecksForFreeCall(final ILocation loc, final RValue pointerToBeFreed,
+			final boolean isPointerCheckRequired, final RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			final MemoryModelDeclarationsHandler memoryModelDeclarationsHandler);
+
+	/**
+	 * Converts a pointer to an int.
+	 *
+	 * @return The new int expression.
+	 */
+	ExpressionResult convertPointerToInt(final ILocation loc, final ExpressionResult rexp, final CPrimitive newType);
+
+	/**
+	 * Converts an int to a pointer.
+	 *
+	 * @return The new pointer expression.
+	 */
+	ExpressionResult convertIntToPointer(final ILocation loc, final ExpressionResult rexp, final CPointer newType);
+
+	/**
+	 * Creates a function pointer, with the given offset.
+	 *
+	 * @return The function pointer.
+	 */
+	Expression createFunctionPointer(final ILocation loc, final BigInteger offset);
+
+	/**
+	 * Adds an expression to a pointer.
+	 *
+	 * @return The new pointer.
+	 */
+	Expression addExpressionToPointer(final ILocation loc, final Expression ptrExpr, final Expression expr);
+
+	/**
+	 * Returns a pointer to the last character of a string.
+	 *
+	 * @return The pointer.
+	 */
+	Expression lastCharOfString(final ILocation loc, final CPrimitive sizeT, final IdentifierExpression len,
+			final IdentifierExpression returnValue);
+
+	/**
+	 * Creates the assume statement used in the handling of strchr.
+	 *
+	 * @return The statement.
+	 */
+	AssumeStatement strChrAssumeStatement(final ILocation loc, final Expression tmpExpr, final Expression argSPtr,
+			final Expression nullPtrExpr, final RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			final MemoryModelDeclarationsHandler memoryModelDeclarationsHandler);
+
+	/**
+	 * Constructs assert / assume statements for ptr memsafety checks.
+	 *
+	 * @return The statements.
+	 */
+	List<Statement> constructMemSafeStatementsForPointerExpression(final ILocation loc, final Expression ptr,
+			final CheckMode pointerBaseValid, final CheckMode pointerTargetFullyAllocated,
+			final RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			final MemoryModelDeclarationsHandler memoryModelDeclarationsHandler);
+
+	/**
+	 * Construct the assume to check that src and dest don't overlap.
+	 *
+	 * @return The assume statement.
+	 */
+	Statement checksForStringCopyOverlapping(final ILocation loc, final Expression src, final Expression srcId,
+			final Expression destId, final Expression dest);
+
+	/**
+	 * Constructs the rhs for the assignment statement of an heap data array.
+	 *
+	 * @return The expression.
+	 */
+	Expression[] rhsAssignmentStatementHda(final ILocation loc, final HeapDataArray hda, final Expression baseAddress);
+
+	/**
+	 * Returns an initial pointer with the same base address. If 2D-Addressing, than the offset is 0.
+	 *
+	 * @return The initial pointer.
+	 */
+	Expression initialPointerFromPointer(final ILocation loc, final Expression ptr);
```

Comment:
Add prefix `construct` to interface method name.

```suggestion
	Expression constructInitialPointerFromPointer(final ILocation loc, final Expression ptr);
```

Reply:
See [bf2a925](https://github.com/ultimate-pa/ultimate/commit/bf2a9251bf23c01929004e0a22a407d1c43a4f33)